contracts/src/tokens/drop/DropFacetBase.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IDropFacetBase} from "./IDropFacet.sol";
+
+// libraries
+import {DropStorage} from "./DropStorage.sol";
+import {CustomRevert} from "contracts/src/utils/libraries/CustomRevert.sol";
+import {MerkleProofLib} from "solady/utils/MerkleProofLib.sol";
+
+// contracts
+
+abstract contract DropFacetBase is IDropFacetBase {
+  using DropStorage for DropStorage.Layout;
+  using MerkleProofLib for bytes32[];
+
+  function _getActiveConditionId(
+    DropStorage.Layout storage ds
+  ) internal view returns (uint256) {
+    uint256 conditionStartId = ds.conditionStartId;
+    uint256 conditionCount = ds.conditionCount;
+
+    for (
+      uint256 i = conditionStartId + conditionCount;
+      i > conditionStartId;
+      i--
+    ) {
+      ClaimCondition memory condition = ds.conditionById[i - 1];
+      if (
+        block.timestamp >= condition.startTimestamp &&
+        (condition.endTimestamp == 0 ||
+          block.timestamp < condition.endTimestamp)
+      ) {
+        return i - 1;
```

Comment:
```suggestion
    if (conditionCount == 0)
      CustomRevert.revertWith(DropFacet__NoActiveClaimCondition.selector);

    for (
      uint256 i = conditionStartId + conditionCount - 1;
      i >= conditionStartId;
      --i
    ) {
      ClaimCondition storage condition = ds.conditionById[i];
      uint256 endTimestamp = condition.endTimestamp;
      if (
        block.timestamp >= condition.startTimestamp &&
        (endTimestamp == 0 || block.timestamp < endTimestamp)
      ) {
        return i;
```
Saves one `sub` and multiple `sload` per loop.

Reply:
this is a view function